### ISSUE TITLE
Should ascii.fixed_width be able to infer column start (or end) positions only from header?

### ISSUE THREAD

**Issue by dhomeier** [astropy/astropy#10488]
### Description
#10481 presented an example of a somewhat exotic table that could be parsed as `fixed_width` but would not be detected automatically. I found however that much simpler examples of tables are not always parsed as I at least expected. Take a file
```
a   b    
1   2.3  
 4  5.6e7
```
that can be read in by setting the `delimiter` option to a blank value like `None` or `' '`, which will however determine both column positions _and widths_ from the header entries:
```
>>> ascii.read(t, format='fixed_width', delimiter=None)
<Table length=2>
  a     b  
int64 int64
----- -----
    1     2
   --     5
```
One may reasonably want to read in the full content of the rows (assuming each column ends just before the start of the next one), but presently this requires explicitly specifying all column positions:
```
>>> ascii.read(t, format='fixed_width', col_starts=[0, 4])
<Table length=2>
  a       b     
int64  float64  
----- ----------
    1        2.3
    4 56000000.0
```
For (more machine- than human-readable) fixed-width tables it is not even uncommon for the content to completely fill the rows
```
a   b    
10242.345678
00425.606e07
```
to be read as
```
<Table length=2>
  a       b     
int64  float64  
----- ----------
 1024   2.345678
   42 56060000.0
```
So what I propose is a modification to `get_fixedwidth_params` allowing it to return the column `starts` only from the header entries (and infer the `ends` from them, just as if the `col_starts` argument had been given). Questions:

1. Would the column start positions indicated by the header names be enough as the typical case, as in the example above, or would we also want a complementary option determining the ends only (i.e. for right-aligned captions)?

2. If the `col_starts` case is enough, would it be acceptable to make this the default for `delimiter=None`? `delimiter=' '` could still reproduce the present behaviour, so the only case that would no longer work that way would be a table requiring the generic 'blank' delimiter (e.g. for a mix of space and `\t` separators).

I have let the `io.ascii` and `table` tests run with the 1. option made default for `delimiter=None` without any failures, so at least in the tests there seems to be no conflicting use of that delimiter setting.

Some tests may make the proposed behaviour clearer:
```
table = """\
| Col1 |  Col2     |  Col3     |  Col4     |
| 1.2  | "hello"   |  1        |  a        |
| 2.4  | 's worlds |         2 |         2 |
"""
dat = ascii.read(table, Reader=ascii.FixedWidth)


def test_read_header_col_starts():
    """Table with blank delimiter with column start positions demarked by header"""
    table = """
  Col1   Col2         Col3        Col4      
  1.2    "hello"      1           a         
  2.4    's worlds           2           2  
"""
    tab = ascii.read(table, Reader=ascii.FixedWidth, delimiter=None)
    assert_equal(tab.colnames, dat.colnames)
    for c in dat.colnames:
        if dat[c].dtype.kind == 'f':
            assert np.allclose(dat[c], tab[c])
        else:
            assert np.array_equal(dat[c], tab[c])

    # Compare delimiter=None (defining start positions) to delimiter=' ' (start and end)
    table = """
Col01  Col02   Col3 
    100  1.2345XA
    200123.4e+12A03b
"""
    tab = ascii.read(table, Reader=ascii.FixedWidth, delimiter=None)
    assert np.array_equal(tab['Col01'], np.arange(1, 3) * 100)
    assert np.allclose(tab['Col02'], np.array([1.2345, 1234]))
    assert np.array_equal(tab['Col3'], np.array(['XA', '2A03b']))

    tab = ascii.read(table, Reader=ascii.FixedWidth, delimiter=' ')
    assert np.array_equal(tab['Col01'], np.arange(1, 3))
    assert np.allclose(tab['Col02'], np.array([1.2, 123.4]))
    assert np.array_equal(tab['Col3'], np.array(['XA', '2A03']))
```

**Comment by taldcroft** [comment]
What about allowing `col_starts` or `col_ends` to take a special value like `'from_header'` that indicates to compute them from the header?  The fixed width reader is already a complex beast so I think it would be useful to be explicit about what's happening instead of making it a special-case default behavior.  This would also prevent inadvertent breakage of existing code.

**Comment by dhomeier** [comment]
> What about allowing `col_starts` or `col_ends` to take a special value like `'from_header'` that

Yes, I had thought about that; the kwargs are directly passed to `self.header.col_starts` etc. and should accept a string value without problems. I was just a little worried about overloading them, but that's acceptable, it is probably the least intrusive solution, plus having the advantage of supporting both `col_starts` and `col_ends`.
I saw some logic why `delimiter=None` should trigger the new behaviour, and `delimiter=' '` or `'\t'` the old one, but it would probably need some good explanation in the fixed-width gallery... ;-)
